app_rstats/main.py

```diff
@@ -753,3 +784,255 @@ def _safe_window_for_geom(dataset, g):
         return result
     except Exception:
         logger.exception("something bad happened")
+
+
+@app.post("/stats/scatter", response_model=ScatterOut)
+def geometry_scatter(q: GeometryScatterIn):
+    try:
+        logging.debug("Opening rasters")
+        dsx, nodata_x, units_x = _open_raster(q.raster_id_x)
+        dsy, nodata_y, units_y = _open_raster(q.raster_id_y)
+
+        logging.debug("Shaping geometry")
+        geom = shape(q.geometry)
+
+        if q.from_crs != dsx.crs.to_string():
+            logging.debug("Reprojecting geometry to X raster CRS")
+            transformer = Transformer.from_crs(
+                q.from_crs, dsx.crs, always_xy=True
+            )
+            geom = shp_transform(
+                lambda x, y, z=None: transformer.transform(x, y), geom
+            )
```

Comment:
check to see if this is necessary, we can just require that the input is the same CRS as the raster thus avoiding a reprojection?

Reply:
it is because it's coming in as epsg 4326 so we gotta transform because we just don't know. that's okay, but did also change naming to be geom_x so it's consistent with geom_y later 35732b1f5e46674ff2093ec6a464d8012fb7d731